clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareFetchBuffer.java

```diff
@@ -59,4 +59,13 @@ public void close() {
             lock.unlock();
         }
     }
+
+    public void add(CompletedFetch completedFetch) {
+        try {
+            lock.lock();
```

Comment:
The pattern to use with the lock is to lock outside the try-catch, and then unlock in the finally. The close method does it right.

Reply:
The FetchBuffer class that this code was derived from was doing it wrong :)